### PR TITLE
AppDelegate clean up

### PR DESCRIPTION
**Pull request by rakaramos**
@RocketChat/iOS

Move dependencies out from `AppDelegate`.
Use `NSNotifications` to react to the app lifecycle and reconnect.

### CODE REVIEW

File: Rocket.Chat/Controllers/Chat/ChatViewController.swift [hunk 1]
```diff
@@ -71,10 +72,18 @@ class ChatViewController: SLKTextViewController {
...
         setupTextViewSettings()
-        
+
+        // TODO: this should really goes into the view model, when we have it
+        NotificationCenter.default.addObserver(self, selector: #selector(ChatViewController.reconnect), name: NSNotification.Name.UIApplicationDidBecomeActive, object: nil)
         SocketManager.addConnectionHandler(token: socketHandlerToken, handler: self)
     }
-    
+
+    @objc private func reconnect() {
```
Comment (rafaelks) [inline Rocket.Chat/Controllers/Chat/ChatViewController.swift]: Why the `@objc` here?

Comment (rakaramos) [inline Rocket.Chat/Controllers/Chat/ChatViewController.swift]: `NSNotification` needs to have a `@objc` selector to properly work.

Comment (rafaelks) [inline Rocket.Chat/Controllers/Chat/ChatViewController.swift]: Yep, but this is just for old version of Xcode. In Xcode 7+ we don't need that.

**Comment by rakaramos** [comment]
@rafaelks updated

**Comment by rafaelks** [comment]
@rakaramos Awesome, thank you! 🚀

**Pull request merged**